core/modules/ccontrol/UserQueryFactory.cc

```diff
@@ -161,30 +152,28 @@ UserQueryFactory::newUserQuery(std::string const& aQuery,
 // parser maturity will have obviously passed what we could do with antlr v2 and we should remove all the
 // antlr v2 parser code from qserv.
 #if 0
-        // parse using antlr v2
-        std::shared_ptr<query::SelectStmt> a2stmt;
+    {
+        auto parser = parser::SelectParser::newInstance(query, parser::SelectParser::ANTLR2);
         try {
-            a2stmt = antlr2NewSelectStmt(query);
-        } catch(parser::ParseException const& e) {
+            parser->setup();
+        } catch (parser::ParseException& e) {
             LOGS(_log, LOG_LVL_DEBUG, "antlr v2 parse exception: " << e.what());
         }
-        if (a2stmt != nullptr) {
+        auto stmt = parser->getSelectStmt();
+        if (stmt != nullptr) {
```

Comment:
I usually do that for `==` because that's easier to overlook. I find `!=` less likely to be written wrong and IMO "A is not nullptr" reads more naturally than "nullptr is not A"

Reply:
Sure, in that case it reads nicer. (tho it seems unusual to me that I wouldn't want to capture the pointer returned but the function for further processing).